### PR TITLE
fix(RHTAPWATCH-819): Resolve EC label errors

### PR DESCRIPTION
**Pull request by hmariset**
Add missing labels which should be present in Dockerfile for the integration tests to pass.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -30,4 +30,13 @@ WORKDIR /
 COPY --from=builder /workspace/manager .
 USER 65532:65532
 
+# It is mandatory to set these labels
+LABEL name="Konflux Project Controller"
+LABEL description="Konflux Project Controller"
+LABEL com.redhat.component="Konflux Project Controller"
+LABEL io.k8s.description="Konflux Project Controller"
+LABEL io.k8s.display-name="o11y-project-controller"
```
Comment (ifireball) [inline Dockerfile:38]: This has nothing to do with o11y. Please use "konflux" here.

**Review by ifireball — CHANGES_REQUESTED** [review]

File: Dockerfile [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # Build the manager binary
-FROM golang:1.21 AS builder
+FROM registry.access.redhat.com/ubi9/go-toolset:1.20.10-2 AS builder
```
Comment (ifireball) [inline Dockerfile:2]: We need Go 1.21.x

**Review by ifireball — CHANGES_REQUESTED** [review]

**Comment by ifireball** [comment]
Please rebase to see if the latest code builds

File: Dockerfile (hunk 2 above)
Comment (hmariset) [reply Dockerfile:2]: We don't have redhat Go 1.21.x image yet. The latest one 1.20.10-3.

**Pull request merged**